### PR TITLE
Deduplicate code for syncing target configmaps and secrets

### PR DESCRIPTION
**Pull request by erikgb**
Currently, the code for syncing target configmaps and secrets is separate - even if the logic should be the same. This doesn't make it tempting to add new features to bundle targets. In this PR I made an attempt to duplicate some of the code. We could probably do more, but let me know what you think before I spend more time on this! ☺️

### CODE REVIEW

File: pkg/bundle/bundle.go [hunk 1]
```diff
@@ -239,169 +237,107 @@ func (b *bundle) reconcileBundle(ctx context.Context, req ctrl.Request) (result
...
 			}
 		}
 	}
 
-	// Find all old existing ConfigMap targetResources.
-	{
-		configMapList := &metav1.PartialObjectMetadataList{
+	// Find all old existing target resources.
+	for _, kind := range []targetKind{configMapTarget, secretTarget} {
+		targetLog := log.WithValues("kind", kind)
```
[the remark this answers is not in the document]
Comment (inteon) [reply pkg/bundle/bundle.go:250]: ( I think )

File: pkg/bundle/bundle.go [hunk 2]
```diff
@@ -201,12 +202,9 @@ func (b *bundle) reconcileBundle(ctx context.Context, req ctrl.Request) (result
...
 
-	type targetResource struct {
-		Kind targetKind
-		types.NamespacedName
-	}
-
-	targetResources := map[targetResource]bool{}
+	targetResources := map[targetKind]map[types.NamespacedName]bool{}
+	targetResources[configMapTarget] = map[types.NamespacedName]bool{}
+	targetResources[secretTarget] = map[types.NamespacedName]bool{}
```
Comment (inteon) [inline pkg/bundle/bundle.go:207]: Why did you change this?

Comment (erikgb) [reply pkg/bundle/bundle.go:207]: Because I now use nested maps. I think it made a lot of sense for the rest of the code, but that requires initialization of the nested maps.

File: pkg/bundle/bundle.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (erikgb) [reply pkg/bundle/bundle.go:250]: Hmmm I am not sure, but it is doable with a condition. This code is somehow protected by https://github.com/cert-manager/trust-manager/blob/main/pkg/bundle/bundle.go#L178-L196, but there is a slight difference now.

[the remark this answers is not in the document]
Comment (erikgb) [reply pkg/bundle/bundle.go:254]: I think you are right. I added a condition now.

File: pkg/bundle/bundle.go (hunk 2 above)
Comment (inteon) [reply pkg/bundle/bundle.go:207]: Sorry, that is what I meant: why are we using a nested map?

Comment (erikgb) [reply pkg/bundle/bundle.go:207]: I tried to answer that already: It does not require creating a custom type and I think it makes the loops more compact and easier to read. But I can revert to the current structure.

Comment (erikgb) [reply pkg/bundle/bundle.go]: Reverted the attempt to use nested maps.

File: pkg/bundle/bundle.go [hunk 3]
```diff
@@ -351,54 +306,40 @@ func (b *bundle) reconcileBundle(ctx context.Context, req ctrl.Request) (result
...
+			syncFunc = func(targetLog logr.Logger, target targetResource, shouldExist bool) (bool, error) {
+				return b.syncSecretTarget(ctx, targetLog, &bundle, target.Name, target.Namespace, resolvedBundle, shouldExist)
 			}
 		}
 
-		if cmSynced || secretSynced {
+		synced, err := syncFunc(targetLog, target, shouldExist)
+		if err != nil {
+			targetLog.Error(err, "failed sync bundle to target namespace")
+			b.recorder.Eventf(&bundle, corev1.EventTypeWarning, fmt.Sprintf("Sync%sTargetFailed", target.Kind), "Failed to sync target %s in Namespace %q: %s", target.Kind, target.Namespace, err)
```
[the remark this answers is not in the document]
Comment (inteon) [reply pkg/bundle/bundle.go:325]: Probably for another PR. Will approve and merge this PR first.

**Comment by inteon** [comment]
Thanks @erikgb. I would like you to continue making improvements, let's do it in separate PRs.

/approve
/lgtm
